src/components/table/column-picker/column-picker.component.ts

```diff
@@ -288,8 +298,8 @@ export class ColumnPickerComponent implements OnChanges {
     }
 
     /** Provide a trackBy function for the reorderable options */
-    selectedTrackBy(index: number, column: string): string {
-        return index + column;
+    selectedTrackBy(index: number, column: string | ColumnPickerGroupItem): string {
+        return index + (column['name'] || column);
```

Comment:
`object['property']` syntax should be avoid wherever possible due to bypassing type checking. Consider implementing something like this: https://stackoverflow.com/a/44078574

```ts
return isColumnPickerGroupItem(column) ? column.name : column;
```

This would be more expressive. There are several places that this could be used, including the tests.